### PR TITLE
Cumulus 735

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Integration tests for changing a workflow and re-deploying while the workflow is running

Addresses [CUMULUS-735](https://bugs.earthdata.nasa.gov/browse/CUMULUS-735), [CUMULUS-736](https://bugs.earthdata.nasa.gov/browse/CUMULUS-736), [CUMULUS-737](https://bugs.earthdata.nasa.gov/browse/CUMULUS-737)

## Changes

* Added utility functions for workflow redeploy
* Added a test lambda that waits for the redeployment to complete
* Test for removing and redeploying a workflow
* Test for updating and redeploying a workflow

## Test Plan
Things that should succeed before merging.

- [ ] Unit tests
- [ ] Adhoc testing
- [ ] Update CHANGELOG
- [ ] Run `./bin/eslint-ratchet` and verify that eslint errors have not increased
  - [ ] Commit `.eslint-ratchet-high-water-mark` if the score has improved

## Release PR

If this is a **release** PR, make sure you branch name starts with `release-` prefix, e.g. `release-v-1.1.2`.

Reviewers: @tag-your-friends

### CODE REVIEW

File: example/spec/redeployment/WorkflowRedeploySpec.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+const {
```
Comment (Jkovarik) [inline example/spec/redeployment/WorkflowRedeploySpec.js]: This file should use "use strict"

File: example/spec/helpers/workflowUtils.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+const yaml = require('js-yaml');
+
+const {
+  get,
+  omit,
+  unset
+} = require('lodash');
+
+const workflowsYmFile = './workflows.yml';
+const workflowsYmlCopyFile = './workflowsCopy.yml';
```
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply example/spec/helpers/workflowUtils.js]: Yep had a typo there (nice catch) and renamed files.

File: example/spec/helpers/testUtils.js [hunk 3]
```diff
@@ -92,22 +92,27 @@ function getExecutionUrl(executionArn) {
...
- * @param {string} workflowName - workflow name
- * @returns {Object} return the workflow configuration
+ * @param {Object} config - configuration object from loadConfig()
+ * @returns {undefined} none
  */
-function getWorkflowConfig(workflowConfigFile, workflowName) {
-  const config = yaml.safeLoad(fs.readFileSync(workflowConfigFile, 'utf8'));
-  if (workflowName) return config[workflowName];
-  return config;
+async function redeploy(config) {
```
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply example/spec/helpers/testUtils.js]: I added a section in the README. We could also configure these tests to go last.

File: example/spec/helpers/workflowUtils.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+ *
+ * @param {string} workflowName - workflow to remove
+ * @param {string} workflowConfigFile - workflow config file, defaults to
+ * workflows.yml
+ * @returns {undefined} none
+ */
+function removeWorkflow(workflowName, workflowConfigFile = workflowsYmFile) {
+  let config = loadWorkflowConfigFile(workflowConfigFile);
+
+  config = omit(config, workflowName);
```
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply example/spec/helpers/workflowUtils.js]: Thanks for the heads up. Made the update.

File: packages/deployment/iam/cloudformation.template.yml [hunk 5]
```diff
@@ -298,6 +298,12 @@ Resources:
                   - sns:List*
                 Resource: '*'
 
+              # Needed for AWS integration tests
```
[the remark this answers is not in the document]
Comment (laurenfrederick) [reply packages/deployment/iam/cloudformation.template.yml]: I have to update IAM roles or I get errors so idk.

File: example/spec/redeployment/WorkflowRedeploySpec.js (hunk 1 above)
Comment (laurenfrederick) [reply example/spec/redeployment/WorkflowRedeploySpec.js]: Thanks!

**Pull request merged**